backend/product-service/db/init.go

```
@@ -0,0 +1,230 @@
+
+		// Begin transaction
+		tx, err := db.Begin()
+		if err != nil {
+			return fmt.Errorf("failed to begin transaction: %w", err)
+		}
+
+		// Execute migration
+		logger.Info("Applying migration", zap.String("version", version), zap.String("file", file))
+		_, err = tx.Exec(string(content))
+		if err != nil {
+			tx.Rollback()
```

golangci-lint (1.64.8)

Error return value of `tx.Rollback` is not checked

(errcheck)